src/libraries/Microsoft.Extensions.Configuration/src/ConfigurationManager.cs

```diff
@@ -259,7 +234,78 @@ IEnumerator IEnumerable.GetEnumerator()
             }
         }
 
-        private class ConfigurationBuilderProperties : IDictionary<string, object>
+        private sealed class ProviderManager : IDisposable
```

Comment:
Comments would be helpful, in particular outlining what thread-safety guarantees are being made and what usage is explicitly unprotected / undefined.